### PR TITLE
feat: new pricing page

### PR DESCRIPTION
**Pull request by paulatulis**
@falsaffa  @ksatx @smf434 

Here's a rough draft of the new pricing page. Please leave comments on what you like/don't like/want to see/etc. 

In the pricing grid, I am having trouble getting the tooltip to position in a way where it doesn't make the page jump around. CSS help needed. 

For the FAQ section, I am having issues with getting the font awesome caret  icon to change onClick. When I console.log, everything acts like it should, but it's not re-rendering the component. Any ideas would be greatly appreciated!

Screen Shot 2020-02-28 at 11 55 51 AM

### CODE REVIEW

**Comment by lottamus** [comment]
Deploy preview for *stoplightio* ready!

Built with commit 4e1f2dfc5074b4b54613b9601ffdb2769493fbb5

https://deploy-preview-1422--stoplightio.netlify.com

**Comment by ksatx** [comment]
I like the blue checks/bulletpoints on the cards.

Under the features section, the spacing between the Team and Enterprise columns looks a little funky. Aside from adjusting margins & padding, you might try just changing "Free" to "Free Tier", to even it out a little.

I would suggest adding a little border-radius to the cards, so that they're consistent with other cards on the site.

**Comment by paulatulis** [comment]
> I like the blue checks/bulletpoints on the cards.
> 
> Under the features section, the spacing between the Team and Enterprise columns looks a little funky. Aside from adjusting margins & padding, you might try just changing "Free" to "Free Tier", to even it out a little.
> 
> I would suggest adding a little border-radius to the cards, so that they're consistent with other cards on the site.

Great suggestions! Thanks for taking a look. Will get these implemented.

**Comment by paulatulis** [comment]
Updated some spacing, but will likely need to continue to make adjustments as we go. 

@smf434 do you think we will need a tool tip for every single feature? This will impact some of the logic, so want to account for that now.

**Comment by paulatulis** [comment]
Updated tooltip and pricing plan box css

Screen Shot 2020-03-02 at 8 32 14 AM

**Comment by paulatulis** [comment]
Action items: 
- plan tiers need to be sticky
- not every item needs a tooltip 
- remove g2 widget
- CTA buttons for each plan card 
- colors for different tiers
- FAQ - 2 columns

**Comment by paulatulis** [comment]
> Action items:
> 
> * plan tiers need to be sticky
> * not every item needs a tooltip
> * remove g2 widget
> * CTA buttons for each plan card
> * colors for different tiers
> * FAQ - 2 columns

@ksatx 
The action items from our morning meeting have been addressed. Please check the Netlify preview and let me know what you think, or we can sync in person if that's easier. Thanks!

**Comment by ksatx** [comment]
Updated cards look great, independent of the color-coding. With the color coding, they look extra great! 

- The "What's Included" section is still a little hard to follow visually. Without the table boundaries, there's nothing to keep your eyes on track when moving from left to right. Since Mark wants to ditch the grid table format, I looked for other companies we've sited as analogs in the past that have an extended features list and are _not_ using visual girds.....and there aren't many. Best suggestion would be something like Monday's [https://monday.com/pricing](https://monday.com/pricing). The colors continue down through the checks, creating a clear association. Having column boundaries but no row boundaries keeps it organized without feeling too grid-like. 
- The column spacing is still a bit funky, it think making "class="sticky flex flex-col pb-0 bg-white" a standard width and then centering the title and checks in their respective columns would help. So I guess all three columns should be as wide as the Enterprise, at minimum. The container padding adds to the imbalanced feeling since the Enterprise column already has extra whitespace to the right. 
- Ditto previous comment about border-radius on cards, add a border-radius to the "What's Included" table for consistency down the page. 
- I know it's a symptom of our normal styling, but the text for the individual features is quite light and difficult to read. Tooltips require concentrated _effort_ to read. A little tooltip inspiration https://akryum.github.io/v-tooltip/#/
- When hovering on Toop Tips, a little tiny arrow pops up to the right, almost like there is a drop-down? But it's not clickable. See the image below, all the way on the right side.

image


FAQ
- I like the two-column layout, but it might look quite lopsided depending on how many FAQs we have and how long the answers are. Another thought is something like **https://www.sketch.com/pricing/** "Common Questions", where the title offsets the questions, and the questions are in one column. Leaving them open looks good, when/if more are added in the future we can re-evaluate open onClick. 
- Consider adding another CTA near the FAQs, perhaps as a subtitle for the section. i.e. "Don't see your question? Talk to a solutions engineer" href= '/demo'.

**Comment by ksatx** [comment]
I put together a version of the color palette for the SuperSide folks. I know you have all this info in the style sheet, but seeing it visually helps me. Might try using some of the lighter purple/green/indigo variations to define the columns/which features go with which plans vertically. https://docs.google.com/presentation/d/1pTAUUT61yNkQXaUvIpssfyudjEfNDgN4oGFvlCwz2Pg/edit#slide=id.g1dc87ade2e_0_18

**Comment by paulatulis** [comment]
> Updated cards look great, independent of the color-coding. With the color coding, they look extra great!
> 
> * The "What's Included" section is still a little hard to follow visually. Without the table boundaries, there's nothing to keep your eyes on track when moving from left to right. Since Mark wants to ditch the grid table format, I looked for other companies we've sited as analogs in the past that have an extended features list and are _not_ using visual girds.....and there aren't many. Best suggestion would be something like Monday's https://monday.com/pricing. The colors continue down through the checks, creating a clear association. Having column boundaries but no row boundaries keeps it organized without feeling too grid-like.
> * The column spacing is still a bit funky, it think making "class="sticky flex flex-col pb-0 bg-white" a standard width and then centering the title and checks in their respective columns would help. So I guess all three columns should be as wide as the Enterprise, at minimum. The container padding adds to the imbalanced feeling since the Enterprise column already has extra whitespace to the right.

This is my main priority, but I am having some issues with finagling the CSS. @falsaffa if you are available tomorrow morning can I ask you some questions?

> * Ditto previous comment about border-radius on cards, add a border-radius to the "What's Included" table for consistency down the page.

added

> * I know it's a symptom of our normal styling, but the text for the individual features is quite light and difficult to read. Tooltips require concentrated _effort_ to read. A little tooltip inspiration https://akryum.github.io/v-tooltip/#/

thank you for this! will make updates

> * When hovering on Toop Tips, a little tiny arrow pops up to the right, almost like there is a drop-down? But it's not clickable. See the image below, all the way on the right side.
> 

> image
> 
Will be solved when I update the style of the tooltip - good catch!
> FAQ
> 
> * I like the two-column layout, but it might look quite lopsided depending on how many FAQs we have and how long the answers are. Another thought is something like **https://www.sketch.com/pricing/** "Common Questions", where the title offsets the questions, and the questions are in one column. Leaving them open looks good, when/if more are added in the future we can re-evaluate open onClick.
I actually think it's okay that it's not 100% symmetrical. I think we should see how many FAQs we end up with, but I kind of prefer the two columns to the one. I'll make updates tomorrow so we can see both ways, though.
> * Consider adding another CTA near the FAQs, perhaps as a subtitle for the section. i.e. "Don't see your question? Talk to a solutions engineer" href= '/demo'.
 
Great point I will add.

**Comment by smf434** [comment]
For CTAs, Free & Team = Get Started, Enterprise = Contact Us

For Pricing top section, see new tab, https://docs.google.com/spreadsheets/d/1dSxt8OJvzbVWSEcwxNQfVo-sk8eG_RUUZaZY4BHmIEs/edit#gid=1430450674

**Comment by ksatx** [comment]
- Remove 'Unlimited Users" from Enterprise. It should build on the previous card, so unlimited users is assumed. 

- @paulatulis I have some ideas for a more gitlab-y/gitbook-y style but they're not very fleshed out and will be easier for me to talk though out loud.  They both use long column cards that are broken into different sections within the cards. I think that looks messy, my thought is breaking into two sets of cards with visual interest between. Some ideas - https://www.canva.com/design/DAD1gRhwpHo/VXHddjoHVMXH7rYNaAEabQ/view#1

That's a big left turn ^ from where we're at now, and the "What's Included" section is already looking so much better, another approach might be:
   - like we already talked about, add column definition to the "What's Included List" and/or plan specific checks (purple check for enterprise, so forth)
   - play with top margins and bullet points to visually represent a good/better/best idea, juxtaposed against the angled hero


image
image



image
 
image

**Comment by paulatulis** [comment]
More updates:
- pricing cards are now incrementally taller, which makes placing a button in the same place on every card kind of tricky since it depends on where the content ends/starts on each one, and now they are all different. I moved it to the very bottom, which is even across, so please let me know what you think. 
- divider lines between sections. I can also use chips, but we've noticed that they are confusing and people try to click on them, so I stuck with lines. 
- used classic FAQ as placeholder so we can see what it looks like with varying question/answer lengths


Please continue to let me know your thoughts. Thanks!

File: src/components/PricingPlans/index.tsx [hunk 1]
```diff
@@ -37,35 +41,44 @@ export const PricingPlan: React.FunctionComponent<IPricingPlan> = ({
...
-          {price}
-          {unit && <span className="font-default text-base ml-3">{unit}</span>}
-        </div>
-
-        <div className="mb-10">
-          {features.map((feature, key) => (
-            <PlanFeature key={key} name={feature} color="green" />
-          ))}
+    <div className={`flex-1 flex flex-col h-full mx-3 rounded-lg md:my-6 md:flex-auto md:w-full bg-${titleColor}`}>
+      <div className="items-start flex-1 p-10 pb-20 mt-2 bg-white shadow-md ">
```
Comment (ksatx) [inline src/components/PricingPlans/index.tsx]:
```suggestion
      <div className="items-start flex-1 p-10 mt-2 bg-white shadow-md ">
```

File: tailwind.config.js [hunk 2]
```diff
@@ -317,6 +317,7 @@ module.exports = {
     '3xl': '1.875rem', // 30px
     '4xl': '2.25rem', // 36px
     '5xl': '2.75rem', // 48px
+    '6xl': '3.5rem', // 48px
```
Comment (ksatx) [inline tailwind.config.js]: The title is still displaying as 5xl on the preview for me, I think this needs to be changed in src/components/Hero/index.tsx 

https://github.com/stoplightio/stoplight.io/blob/20b8fb509fea655cb3489f8e4af17039ccb434ad/src/components/Hero/index.tsx#L164

**Comment by ksatx** [comment]
- Leverage the bullet point spacing to achieve the desired height on each card and distribute the negative space more evenly across top and bottom, rather than having it all at the bottom of the list
- let's get clarification from @smf434 tomorrow on the features per plan & see if anything changed as an outcome of today's pricing meeting, that might affect bullets. 
- I actually like the button being sticky to the bottom of the card & full width, that looks nice
- still feel the need for some kind of vertical *or* horizontal definition on the table even just alternating shading on the rows like Product Board https://www.productboard.com/pricing/
- I _think_ I put the suggestion in the right place, but just to CMA, there is padding-bottom on top of padding on the card body, removing that extra padding-bottom cuts down on empty space

image

File: tailwind.config.js (hunk 2 above)
Comment (paulatulis) [reply tailwind.config.js]: You're right! I was just being a scaredy cat in changing every single one, but will do it and we can take a look at the site holistically to make sure it looks good.

File: src/components/PricingPlans/index.tsx (hunk 1 above)
Comment (paulatulis) [reply src/components/PricingPlans/index.tsx]: Made update directly in code, so resolving this.

**Comment by paulatulis** [comment]
Pushed up most recent changes, but given the feature breakdown might change again, I am holding on continuing to work on it at the moment. 

Still to do:
- think about design of top cards. Number of bullets on each card doesn't jive with the way they are laid out, but this might change again
- tooltip styling
- make social proof look better
- table alternate line fill logic

**Comment by paulatulis** [comment]
[Resurrect Pricing page](https://trello.com/c/1KGdIsy9/413-resurrect-pricing-page)

**Comment by paulatulis** [comment]
[Update Pricing page content](https://trello.com/c/1YkWEUOS/414-update-pricing-page-content)

**Comment by paulatulis** [comment]
Tooltip is now appearing above the question mark on hover. 

I'm not sure how much this css can be pared down/optimized, so if anyone has suggestions, let me know :) 
Screen Shot 2020-03-06 at 12 04 45 PM

**Comment by ksatx** [comment]
I don't have any useful fixes to offer, but the tooltip is showing up off-center for me, and the alternating row colors get funky between sections.  
image

image

**Comment by paulatulis** [comment]
> I don't have any useful fixes to offer, but the tooltip is showing up off-center for me, and the alternating row colors get funky between sections.
> image
> 
> image

I'll keep playing around with the tooltip issue. Line colors were funky because of the way the yaml file was constructed and I just pushed up changes to fix that.

**Comment by paulatulis** [comment]
Screen Shot 2020-03-06 at 5 15 03 PM
Pricing:

1. Spacing between plan bullet points can be reduced - Done
2.  Tooltip - use dotted line instead of question mark - Done
3.  Don’t need shadow under grey line in sticky plan breakdown box - Done
4.  Add CTA below entire table - Done but looks repetitive with the one at the bottom of the page as well. @ksatx and I can brainstorm some other options.
5.  Border-b on top plan card sections - Done
6.  One line item for integrations in Pricing grid - Done
7.  Tooltip - use popper.js
8.  Vertical alignment needs attention 
9.  Top of card - reduce padding 
10.  Try page with curved hero background
11.  Reduce padding all around 
12.  Difficult to understand some of the pricing plan items, could put anchor links to the part of the features matrix that has the tooltip for that bullet point - Wait on final copy and bullet points to do this
13.  Instead of white label, ‘remove stoplight logo’ (? I am not sure what this note means. @ksatx do you remember?)

**Comment by ksatx** [comment]
@paulatulis - here, changing "white labeled"  to "Remove Stoplight Logo" - it's less ambiguous
image

**Comment by paulatulis** [comment]
Just made some updates to styling for medium and small screens. Still waiting on some content-related items. The only other big thing on my list is tooltips. 

Please see running list and let me know what else needs attention. 

**Hero**

1. Copy for H1: 

 - Pricing to Fit Your Needs
 - Stoplight Pricing and Features
 - Choose your Plan
 

**Pricing Plan Cards**
1. Colors for plans
2. Final bullets for each plan 
3. Links to CTA for each plan 
  - Enterprise: demo form
  - Team:
  - Free:

**Logo Collage**
1. Stretch goal: animation 

**Pricing Matrix**
1. Tooltip position - **help**
2. Final bullets and tooltip copy

**Action Bar**
1. Is placement okay?
2. Confirm copy and CTA buttons

**FAQs**
1. Final FAQ content

**Comment by ksatx** [comment]
First of all, **I'm in love with the color-coded checks.** 


> **Hero**
> 
> 1. Copy for H1:
> 
> * Pricing to Fit Your Needs
> * Stoplight Pricing and Features
> * Choose your Plan

Either "Stoplight Pricing and Features" or "Stoplight Plans and Pricing". Salt might have an opinion after it's gone live & been crawled. 

> 
> **Pricing Matrix**
> 
> 1. Tooltip position - **help**
> 2. Final bullets and tooltip copy
> 

image

Where do you image the tooltip going? I think it's kind of convenient to have it right below the feature, and I also like that it doesn't cover the checks. Tooltips that obscure the original text I was looking at/other important text on the line make me frustrated. 

image

Still seeing border-b on feature matrix. Reducing to 1 px looks good on my end, provides separation but not as intense.  

> **Action Bar**
> 
> 1. Is placement okay?
> 2. Confirm copy and CTA buttons
> 

Let's change it to "Speak with a Solutions Engineer." 

It looks a little odd being so much wider than the feature matrix, more like it's the beginning of a different idea. Should feel like a logical next step if you get to the end and still need help. 

I dropped the max-width, reduced the padding, and rearranged the text.

image

**Comment by paulatulis** [comment]
> First of all, **I'm in love with the color-coded checks.**
> 
yay!
> > **Hero**
> > 
> > 1. Copy for H1:
> > 
> > 
> > * Pricing to Fit Your Needs
> > * Stoplight Pricing and Features
> > * Choose your Plan
> 
> Either "Stoplight Pricing and Features" or "Stoplight Plans and Pricing". Salt might have an opinion after it's gone live & been crawled.
> 
Went with Stoplight Plans and Pricing for now
> > **Pricing Matrix**
> > 
> > 1. Tooltip position - **help**
> > 2. Final bullets and tooltip copy
> 
> image
> 
> Where do you image the tooltip going? I think it's kind of convenient to have it right below the feature, and I also like that it doesn't cover the checks. Tooltips that obscure the original text I was looking at/other important text on the line make me frustrated.
> 
Not too tied to where, but there needs to be a polygon coming out of the message box like so: 
Screen Shot 2020-03-10 at 1 57 17 PM

> image
> 
> Still seeing border-b on feature matrix. Reducing to 1 px looks good on my end, provides separation but not as intense.
> 
border b is now thinner
> > **Action Bar**
> > 
> > 1. Is placement okay?
> > 2. Confirm copy and CTA buttons
> 
> Let's change it to "Speak with a Solutions Engineer."
> 
> It looks a little odd being so much wider than the feature matrix, more like it's the beginning of a different idea. Should feel like a logical next step if you get to the end and still need help.
> 
> I dropped the max-width, reduced the padding, and rearranged the text.
> 
> image
Updated

**Comment by paulatulis** [comment]
closing because we have https://github.com/stoplightio/stoplight.io/pull/1431
